living_documentation_generator/generator.py

```diff
@@ -0,0 +1,316 @@
+        gh_repo_issues: dict[str, list[Issue]] = self._mine_github_issues()
+        # Note: got dict of list of issues for each repository (key is repository id)
+
+        # Data mine GitHub project's issues
+        gh_project_issues = self._mine_github_project_issues()
+
+        # Consolidate all issues data together
+        projects_issues = self._consolidate_issues_data(gh_repo_issues, gh_project_issues)
+
+        # Generate markdown pages
+        self._generate_markdown_pages(projects_issues)
+
+    def _clean_output_directory(self):
+        if os.path.exists(self.output_path):
+            shutil.rmtree(self.output_path)
+        os.makedirs(self.output_path)
+
+    def _mine_github_issues(self) -> dict[str, list[Issue]]:
+        issues = {}
+
+        # Mine issues from every config repository
+        for config_repository in self.repositories:
+            repository_id = f"{config_repository.owner}/{config_repository.name}"
+
+            if GithubManager().store_repository(repository_id) is None:
+                return {}
+
+            logging.info(f"Downloading issues from repository `{config_repository.owner}/{config_repository.name}`.")
+
+            # Load all issues from one repository (unique for each repository)
+            repo_issues = GithubManager().fetch_issues(labels=config_repository.query_labels)
+            for repo_issue in repo_issues:
+                repo_issue.repository = repository_id
+
+            # save list of repository issue under its repository id
+            # Note: not creating OOP object instead if this dict as Issue type will be replaced in next PR
+            issues[repository_id] = repo_issues
+
+        return issues
+
+    def _mine_github_project_issues(self) -> dict[str, ProjectIssue]:
+        if not self.project_state_mining_enabled:
+            logging.info("Project data mining is not allowed. The process will not start.")
+            return {}
+
+        logging.info("Project data mining allowed, starting the process.")
+
+        # Mine project issues for every repository
+        project_issues = {}
+
+        for config_repository in self.repositories:
+            repository_id = f"{config_repository.owner}/{config_repository.name}"
+
+            if GithubManager().store_repository(repository_id) is None:
+                return {}
+
+            # Fetch all projects_buffer attached to the repository
+            projects = GithubManager().fetch_repository_projects(self.projects_title_filter)
+
+            # Update every project with project issue related data
+            for project in projects:
+                for prj_issue in GithubManager().fetch_project_issues(project):
+                    key = make_issue_key(prj_issue.organization_name, prj_issue.repository_name, prj_issue.number)
+                    project_issues[key] = prj_issue
+
+        return project_issues
+
+    def _consolidate_issues_data(self, gh_repo_issues: dict[str, list[Issue]],
```

Comment:
Should be a static method.

Reply:
Let's make it part of our project format rules for similar situations. Addressed in commit - TBF.